### ISSUE TITLE
Adjust class labeling pipeline

### ISSUE THREAD

**Issue by nclgbd** [nclgbd/ResearchToolKit#9]
# Overview

Since the way we're splitting the data is becoming more and more complex, I'd like to change the way we do indexing of the dataset. To do this I want to use [scikit learn's multiclass capabilities](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html#sklearn.preprocessing.MultiLabelBinarizer). A way to quantify each label at different granularities should be implemented. Like statistics on it's own class distribution

## Subtasks

- [x] Add way to report class distributions for each class
- [x] Use `scikit-learn`'s multiclass functionality to transform data instead of whatever I'm doing currently

## Additional sources

https://scikit-learn.org/stable/modules/multiclass.html
https://scikit-learn.org/stable/modules/preprocessing_targets.html#preprocessing-targets
https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html
